### PR TITLE
Sync: Update User Sync Module

### PR DESCRIPTION
**Pull request by lezama**
This PR removes duplicate synced actions.

To test:
With this diff applied try

adding a new user
editing a user
adding / editing via API
editing your profile
add existing user to multi-site
invite user -> user accepts invite
user registers via registration form
delete user
Ensure stuff is syncing properly upstream.

### CODE REVIEW

File: sync/class.jetpack-sync-module-users.php [hunk 1]
```diff
@@ -195,70 +188,40 @@ public function deleted_user_handler( $deleted_user_id, $reassigned_user_id = ''
...
-			 *
-			 * @since 4.9.0
-			 *
-			 * @param object The WP_User object
-			 */
-			do_action( 'jetpack_sync_register_user', $user );
-
-			return;
-		}
+		do_action( 'jetpack_sync_register_user', $user_id );
```
Comment (enejb) [inline sync/class.jetpack-sync-module-users.php]: Do we need to add the Documentation here?

File: sync/class.jetpack-sync-module-users.php [hunk 2]
```diff
@@ -292,66 +251,50 @@ function save_user_role_handler( $user_id, $role, $old_roles = null ) {
...
 
+		if ( isset( $this->flags[ $user_id ] ) ) {
 			/**
 			 * Fires when the client needs to sync an updated user
 			 *
 			 * @since 4.2.0
 			 *
 			 * @param object The Sanitized WP_User object
-		     * @param array state Since 5.8
+			 * @param array state Since 5.8
```
Comment (enejb) [inline sync/class.jetpack-sync-module-users.php]: lets update this to 5.8.0

File: sync/class.jetpack-sync-module-users.php [hunk 3]
```diff
@@ -195,70 +180,52 @@ public function deleted_user_handler( $deleted_user_id, $reassigned_user_id = ''
...
-			 * @param object The WP_User object
-			 */
-			do_action( 'jetpack_sync_register_user', $user );
-
-			return;
-		}
+		/**
+		 * Fires when a new user is registered on a site
+		 *
+		 * @since 4.9.0
```
Comment (jeherve) [inline sync/class.jetpack-sync-module-users.php]: Thank you for adding documentation! Could you also add the `@module sync` text to you Sync hooks, so it gets categorized by the parser?
https://developer.jetpack.com/module/sync/

Thank you!

Comment (lezama) [reply sync/class.jetpack-sync-module-users.php]: noted, do you have time for a quick PR to add the missing ones? 😊

Comment (jeherve) [reply sync/class.jetpack-sync-module-users.php]: Not right now, no, unfortunately, but I'll keep it in mind :)

**Pull request merged**